torch/csrc/lazy/core/tensor_impl.cpp

```diff
@@ -129,7 +146,8 @@ void LTCTensorImpl::shallow_copy_from(
 int64_t LTCTensorImpl::size(int64_t d) const {
```

Comment:
We probably need a mechanism to enable the user to query "true" sizes; something like `int(a.size())`. Of course this triggers a size computation. We currently have a builtin API for this in PT/XLA called `[_get_xla_tensor_dimension_size](https://github.com/pytorch/xla/blob/master/torch_xla/csrc/init_python_bindings.cpp#L790-L793)`.

I know it's not part of the design we discussed with Ed. Would that be something to consider here?

CC @Krovatkin @ezyang @JackCaoG

Reply:
Yes, I think we should add some similar API to LTC as well.

Reply:
Adding this [respond](https://pytorch.slack.com/archives/C03DF2D7YSD/p1651688193843109) from @Krovatkin to the thread here.